test/Extensions/TesterAzureUtils/AzureCosmosDBMembershipTableTests.cs

```diff
@@ -0,0 +1,102 @@
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using TestExtensions;
+using UnitTests.MembershipTests;
+using Xunit;
+using Orleans;
+using Orleans.Messaging;
+using Orleans.Clustering.CosmosDB;
+
+namespace Tester.AzureUtils;
+
+/// <summary>
+/// Tests for operation of Orleans Membership Table using Azure CosmosDB - Requires access to external Azure CosmosDB account
+/// </summary>
+[TestCategory("Membership"), TestCategory("AzureCosmosDB")]
+public class AzureCosmosDBMembershipTableTests : MembershipTableTestsBase
+{
+    public AzureCosmosDBMembershipTableTests(TestEnvironmentFixture environment) : base(environment, CreateFilters())
+    {
+    }
+
+    private static LoggerFilterOptions CreateFilters()
+    {
+        var filters = new LoggerFilterOptions();
+        filters.AddFilter(typeof(AzureCosmosDBMembershipTable).FullName, LogLevel.Trace);
+        filters.AddFilter("Orleans.Storage", LogLevel.Trace);
+        return filters;
+    }
+
+    protected override IMembershipTable CreateMembershipTable(ILogger logger)
+    {
+        var options = new AzureCosmosDBClusteringOptions();
+        options.ConfigureTestDefaults();
+        return new AzureCosmosDBMembershipTable(loggerFactory, this.Services, Options.Create(options), this.clusterOptions);
+    }
+
+    protected override IGatewayListProvider CreateGatewayListProvider(ILogger logger)
+    {
+        var options = new AzureCosmosDBClusteringOptions();
+        options.ConfigureTestDefaults();
+        return new AzureCosmosDBGatewayListProvider(loggerFactory, this.Services, Options.Create(options), this.clusterOptions, this.gatewayOptions);
+    }
+
+    protected override Task<string> GetConnectionString()
+    {
+        return Task.FromResult(TestDefaultConfiguration.DataConnectionString);
+    }
+
+    [SkippableFact, TestCategory("Functional")]
+    public void MembershipTable_Azure_Init()
+    {
+    }
+
+    [SkippableFact, TestCategory("Functional")]
+    public async Task MembershipTable_Azure_GetGateways()
+    {
+        await MembershipTable_GetGateways();
+    }
+
+    [SkippableFact, TestCategory("Functional")]
+    public async Task MembershipTable_Azure_ReadAll_EmptyTable()
+    {
+        await MembershipTable_ReadAll_EmptyTable();
+    }
+
+    [SkippableFact, TestCategory("Functional")]
+    public async Task MembershipTable_Azure_InsertRow()
+    {
+        await MembershipTable_InsertRow();
+    }
+
+    [SkippableFact, TestCategory("Functional")]
```

Comment:
This way of testing is new to me? Is there any docs guidance on it?

Just wondering if the tests might read a little better if we was to follow a standard naming convention and maybe not have test's calling test methods? Happy to help support a potential re-structure of this :)

Reply:
I've followed what other tests are doing.